themes/pom.xml

```diff
@@ -11,6 +11,115 @@
+                        <phase>initialize</phase>
+                        <goals>
+                            <goal>npm</goal>
+                        </goals>
+                        <configuration>
+                            <workingDirectory>${dir.account2}</workingDirectory>
+                            <arguments>${args.npm.install}</arguments>
+                        </configuration>
+                    </execution>
+                </executions>
+                <configuration>
+                    <nodeVersion>v12.13.0</nodeVersion>
+                    <npmVersion>6.9.0</npmVersion>
+                    <installDirectory>node</installDirectory>
+                </configuration>
+            </plugin>
+        </plugins>
+
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+                <excludes>
+                    <exclude>**/node_modules/**/node_modules/**</exclude>
+                    <exclude>**/minimist/**</exclude>
+                    <exclude>**/mkdirp/**</exclude>
+                    <exclude>**/package.json</exclude>
+                    <exclude>**/bower.json</exclude>
+                    <exclude>**/component.json</exclude>
+                    <exclude>**/composer.json</exclude>
+                    <exclude>**/npm-shrinkwrap.json</exclude>
+                    <exclude>**/select2.jquery.json</exclude>
+                    <exclude>**/*.markdown</exclude>
+                    <exclude>**/*.swf</exclude>
+                    <exclude>**/*.sh</exclude>
+                    <exclude>**/.bin/**</exclude>
+                    <exclude>**/bin/**</exclude>
+                    <exclude>**/build/**</exclude>
+                    <exclude>**/docs/**</exclude>
+                    <exclude>**/demo/**</exclude>
+                    <exclude>**/devtools/**</exclude>
+                    <exclude>**/example/**</exclude>
+                    <exclude>**/examples/**</exclude>
+                    <exclude>**/grunt/**</exclude>
+                    <exclude>**/less/**</exclude>
+                    <exclude>**/sass/**</exclude>
+                    <exclude>**/scss/**</exclude>
+                    <exclude>**/jquery/src/**</exclude>
+                    <exclude>**/angular-treeview/src/**</exclude>
+                    <exclude>**/test/**</exclude>
+                    <exclude>**/tests/**</exclude>
+                    <exclude>**/_config.yml</exclude>
+                    <exclude>**/api.md</exclude>
+                    <exclude>**/AUTHORS.txt</exclude>
+                    <exclude>**/CHANGELOG.md</exclude>
+                    <exclude>**/CONTRIBUTING.md</exclude>
+                    <exclude>**/HELP-US-OUT.txt</exclude>
+                    <exclude>**/README.md</exclude>
+                    <exclude>**/Gruntfile.js</exclude>
+                    <exclude>**/Gemfile*</exclude>
+                    <exclude>**/index.js</exclude>
```

Comment:
I'd remove this if possible. It seems more dependencies (not just patternfly/react) rely on an `index.js` (e.g. exenv) and without it the Account Console doesn't work.

Reply:
Including all the index.js files will introduce about 13M into the distribution, these aren't super light files for some reason.

Reply:
Yes, but this way we'll need to manually pick those index.js files that are really required by some of our (transitive) dependencies. And that is IMHO quite a big hassle, mainly when e.g. updating dependencies or introducing new ones. Every time we'll need to manually investigate whether this or that index.js is really required.

Reply:
Thinking about this more. Maybe this is more of a generic question what to really include in the distribution. Couldn't this be somehow derived from [systemjs.config.js](https://github.com/keycloak/keycloak/blob/master/themes/src/main/resources/theme/keycloak-preview/account/resources/systemjs.config.js)? @ssilvert?

npm surely downloads lots of "garbage" and we need only a few files.

On the other hand this is basically an optimization task. Couldn't this be perhaps done after the tech preview?

Reply:
We can do the optimization later, yeah.